types/karma/index.d.ts

```diff
@@ -700,6 +692,31 @@ export interface BrowserConsoleLogOptions {
     terminal?: boolean;
 }
 
+interface ParseOptions {
+    /**
+     * When true, the return value of the function is a Promise of Config instead of Config.
+     * Should be set to true to support async Karma configuration file.
+     *
+     * @deprecated Will become a default in the next major release.
+     */
+    promiseConfig: boolean;
+
+    /**
+     * When true, function will throw error or return rejected Promise instead of calling process.exit(1).
+     *
+     * @deprecated Will become a default in the next major release.
+     */
+    throwErrors: boolean;
+}
+
 export namespace config {
     function parseConfig(configFilePath: string, cliOptions: ConfigOptions): Config;
+    // The return value is technically not correct as user can pass options and set promiseConfig to false.
+    // But from the pragmatic point of view, nobody should pass it like that, so it is better to provide a useful,
+    // although not technically correct return type.
```

Comment:
The default is actually false, so we should really support the correct return type if we can. You can have two overloads, one which has it set to true and the other with it set to false. Or a conditional return type. Whichever is easier

Reply:
someting like (pseudo-code)
...
function parseConfig(configFilePath: string, cliOptions: ConfigOptions): Config;
function parseConfig(configFilePath: string, cliOptions: ConfigOptions & { promiseConfig: true }): Promise<Config>;
```
We can't bind to `promiseConfig?: true;
```
as there is probably existing code that uses `false`, like:
https://github.com/karma-runner/karma/pull/3660/files#diff-37b88258822cb207e2caa1a88d4fbc057924da72c5580d1f39b78349214ab622R58

Reply:
Well, that code is only in Karma itself for backward compatibility. I don't think any Karma consumers will actually do that as the behavior is deprecated and there is no reason to use it over omitting the parameters altogether. But in any case, the main reason was that I didn't know how to make the correct return type 😄 Thanks for the tip @peterblazejewicz! I've updated the PR.